### PR TITLE
fix generate:model --plugin path + little refacto of code

### LINKED ISSUE #3889: Wrong directory with generate:model --plugin
**Describe the bug**
Run command:  `strapi generate:model userinfo --plugin foo`
You get `project/plugins/foo/models/models/userinfo`

**Expected behavior**
Should be `project/plugins/foo/models/userinfo`

**System**
- Node.js version: 11.11.0
- NPM version: 6.11.1
- Strapi version: 3.0.0-beta.15
- Database: Mongo
- Operating system: macOS

### PR DESCRIPTION
**Pull request by petersg83**
fix #3889 

#### Description of what you did:
- changed path
- refacto in every strapi-generate-* to use similar code

### CODE REVIEW

**Comment by petersg83** [comment]
Thanks !

- You're right, I can add deburr. I replaced it to do exactly like in `strapi-generate-api` where `name` is used instead of directly `scope.id`.
- Same as above, to do like in `strapi-generate-api`
- I'm would be glad to talk about it. How is the naming supposed to be? Many things were inconsistent depending on the package so I'll be happy to establish which one is the good way

File: packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/createUid.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import slugify from '@sindresorhus/slugify';
-
-const nameToSlug = name => slugify(name, { separator: '-' });
+import { nameToSlug } from 'strapi-utils';
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/createUid.js]: Strapi-utils isn't meant to be used on the front-end :) Can you reset those changes :)

File: packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/forms.js [hunk 2]
```diff
@@ -3,9 +3,10 @@ import * as yup from 'yup';
 import { get, isEmpty, toLower, trim, toNumber } from 'lodash';
 import { translatedErrors as errorsTrads } from 'strapi-helper-plugin';
 import { FormattedMessage } from 'react-intl';
+import { nameToSlug } from 'strapi-utils';
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/forms.js]: same here :)

File: packages/strapi-plugin-content-type-builder/package.json [hunk 3]
```diff
@@ -8,7 +8,6 @@
     "description": "content-type-builder.plugin.description"
   },
   "dependencies": {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-type-builder/package.json]: Will need to stay for the front-end ;)

File: packages/strapi-generate-model/lib/before.js [hunk 4]
```diff
@@ -34,41 +35,32 @@ module.exports = (scope, cb) => {
...
   }
 
+  // Format `id`.
+  const name = scope.name || nameToSlug(scope.id);
+
   // `scope.args` are the raw command line arguments.
   _.defaults(scope, {
-    id: _.trim(_.deburr(scope.id)),
+    name,
     idPluralized: pluralize.plural(_.trim(_.deburr(scope.id))),
```
Comment (alexandrebodin) [inline packages/strapi-generate-model/lib/before.js]: Now that `nameToCollectionName`  is in strapi-utils you can use it to create the collectionName / idPluralized. you can refactor to get ride of idPluralized and only use collectionName ;)

**Pull request merged**